imap/index.c

```diff
@@ -5288,60 +5284,20 @@ static int login(struct backend *s __attribute__((unused)),
     return 0;
 }
 
-static int ping(struct backend *s,
+static int ping(struct backend *s __attribute__((unused)),
                 const char *userid __attribute__((unused)))
 {
-    struct backend_ctx *ctx = (struct backend_ctx *) s->context;
-    size_t hostlen = strcspn(s->hostname, "/");
-    hdrcache_t resp_hdrs = NULL;
-    struct body_t resp_body;
-    unsigned statuscode = 0;
-    const char *errstr;
-    int r = 0;
-
-    prot_printf(s->out,
-                "HEAD %s %s\r\n"
-                "Host: %.*s\r\n"
-                "User-Agent: Cyrus/%s\r\n"
-                "Connection: Keep-Alive\r\n"
-                "Keep-Alive: timeout=%u\r\n"
-                "Accept: text/plain\r\n"
-                "\r\n",
-                ctx->path, HTTP_VERSION,
-                (int) hostlen, s->hostname, CYRUS_VERSION, ctx->timeout);
-    prot_flush(s->out);
-
-    /* Read response(s) from backend until final response or error */
-    do {
-        resp_body.flags = BODY_DISCARD;
-        r = http_read_response(s, METH_HEAD, &statuscode,
-                               &resp_hdrs, &resp_body, &errstr);
-        if (r) {
-            break;
-        }
-    } while (statuscode < 200);
-
-    if (r) {
-        syslog(LOG_DEBUG, "extractor_ping: HEAD %s: failed (%s): %s",
-               ctx->path, error_message(r), errstr);
-    }
-    else {
-        syslog(LOG_DEBUG, "extractor_ping: HEAD %s: got status %u",
-               ctx->path, statuscode);
-    }
-
-    if (resp_hdrs) spool_free_hdrcache(resp_hdrs);
-
-    return r;
+    return 0;
```

Comment:
I decided the ping was pointless - it doesn't tell us anything about the actual backend, so it's just a callout that always works.